integration-tests/actions/vrf/vrfv2plus/contract_steps.go

```diff
@@ -652,3 +652,63 @@ func CancelSubsAndReturnFunds(ctx context.Context, vrfContracts *vrfcommon.VRFCo
 		}
 	}
 }
+
+func FundWrapperConsumer(
+	sethClient *seth.Client,
+	subFundingType string,
+	linkToken contracts.LinkToken,
+	wrapperConsumer contracts.VRFv2PlusWrapperLoadTestConsumer,
+	vrfv2PlusConfig *vrfv2plusconfig.General,
+	l zerolog.Logger,
+) error {
+	switch vrfv2plusconfig.BillingType(subFundingType) {
+	case vrfv2plusconfig.BillingType_Link:
+		//fund consumer with Link
+		linkAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(*vrfv2PlusConfig.WrapperConsumerFundingAmountLink))
+		l.Info().
+			Str("Link Amount", linkAmount.String()).
+			Str("WrapperConsumerAddress", wrapperConsumer.Address()).Msg("Funding WrapperConsumer with Link")
+		err := linkToken.Transfer(
+			wrapperConsumer.Address(),
+			linkAmount,
+		)
+		if err != nil {
+			return err
+		}
+	case vrfv2plusconfig.BillingType_Native:
+		//fund consumer with Eth (native token)
+		_, err := actions.SendFunds(l, sethClient, actions.FundsToSendPayload{
+			ToAddress:  common.HexToAddress(wrapperConsumer.Address()),
+			Amount:     conversions.EtherToWei(big.NewFloat(*vrfv2PlusConfig.WrapperConsumerFundingAmountNativeToken)),
+			PrivateKey: sethClient.PrivateKeys[0],
+		})
+		if err != nil {
+			return err
+		}
+	case vrfv2plusconfig.BillingType_Link_and_Native:
+		//fund consumer with Link
+		linkAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(*vrfv2PlusConfig.WrapperConsumerFundingAmountLink))
+		l.Info().
+			Str("Link Amount", linkAmount.String()).
+			Str("WrapperConsumerAddress", wrapperConsumer.Address()).Msg("Funding WrapperConsumer with Link")
+		err := linkToken.Transfer(
```

Comment:
Nit: Can we abstract out fundingLink and fundingNative into separate methods and re-use them in different case statements here?

Reply:
but its already abstracted away to bare minimum.

how would you propose to structure the code here?

Reply:
How about
```go
func FundWrapperConsumer(
	sethClient *seth.Client,
	subFundingType string,
	linkToken contracts.LinkToken,
	wrapperConsumer contracts.VRFv2PlusWrapperLoadTestConsumer,
	vrfv2PlusConfig *vrfv2plusconfig.General,
	l zerolog.Logger,
) error {
	isLink := false
	isNative := false
	switch vrfv2plusconfig.BillingType(subFundingType) {
	case vrfv2plusconfig.BillingType_Link:
		isLink = true
	case vrfv2plusconfig.BillingType_Native:
		isNative = true
	case vrfv2plusconfig.BillingType_Link_and_Native:
		isLink = true
		isNative = true
	default:
		return fmt.Errorf("invalid billing type: %s", subFundingType)
	}
	if isLink {
		//fund consumer with Link
		linkAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(*vrfv2PlusConfig.WrapperConsumerFundingAmountLink))
		l.Info().
			Str("Link Amount", linkAmount.String()).
			Str("WrapperConsumerAddress", wrapperConsumer.Address()).Msg("Funding WrapperConsumer with Link")
		err := linkToken.Transfer(
			wrapperConsumer.Address(),
			linkAmount,
		)
		if err != nil {
			return err
		}
	}
	if isNative {
		//fund consumer with Eth (native token)
		_, err := actions.SendFunds(l, sethClient, actions.FundsToSendPayload{
			ToAddress:  common.HexToAddress(wrapperConsumer.Address()),
			Amount:     conversions.EtherToWei(big.NewFloat(*vrfv2PlusConfig.WrapperConsumerFundingAmountNativeToken)),
			PrivateKey: sethClient.PrivateKeys[0],
		})
		if err != nil {
			return err
		}
	}
	return nil
}
```

I think the issue is that there's a lot of code in getting the right data, making it look very repetitive and copy pasted when it's actually just 1 function call.

Alternatively you could define two closures and use it in the switch cases.

Reply:
ok, this looks nicer, but not too much :)

Reply:
Alternatively you could use closures
```go
func FundWrapperConsumer(
	sethClient *seth.Client,
	subFundingType string,
	linkToken contracts.LinkToken,
	wrapperConsumer contracts.VRFv2PlusWrapperLoadTestConsumer,
	vrfv2PlusConfig *vrfv2plusconfig.General,
	l zerolog.Logger,
) error {
	fundConsumerWithLink := func() error {
		//fund consumer with Link
		linkAmount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(*vrfv2PlusConfig.WrapperConsumerFundingAmountLink))
		l.Info().
			Str("Link Amount", linkAmount.String()).
			Str("WrapperConsumerAddress", wrapperConsumer.Address()).Msg("Funding WrapperConsumer with Link")
		return linkToken.Transfer(
			wrapperConsumer.Address(),
			linkAmount,
		)
	}
	fundConsumerWithNative := func() error {
		//fund consumer with Eth (native token)
		_, err := actions.SendFunds(l, sethClient, actions.FundsToSendPayload{
			ToAddress:  common.HexToAddress(wrapperConsumer.Address()),
			Amount:     conversions.EtherToWei(big.NewFloat(*vrfv2PlusConfig.WrapperConsumerFundingAmountNativeToken)),
			PrivateKey: sethClient.PrivateKeys[0],
		})
		return err
	}
	var err error = nil
	switch vrfv2plusconfig.BillingType(subFundingType) {
	case vrfv2plusconfig.BillingType_Link:
		err = fundConsumerWithLink()
	case vrfv2plusconfig.BillingType_Native:
		err = fundConsumerWithNative()
	case vrfv2plusconfig.BillingType_Link_and_Native:
		err = fundConsumerWithLink()
		if err != nil {
			return err
		}
		err = fundConsumerWithNative()
	default:
		return fmt.Errorf("invalid billing type: %s", subFundingType)
	}
	return err
}
```

Reply:
> ok, this looks nicer in a way, but it also looks a bit of over engineering.

It's just good practice to not copy paste the same code. Just in case there's a bug, and we forget to update one of them. Just that this code block is small, so it looks like over engineering.

Reply:
> It's just good practice to not copy paste the same code.

your words are wisdom!